src/core/analysis-cache.ts

```diff
@@ -113,11 +113,23 @@ export class AnalysisCache {
   }
 
   toString() {
-    return `<AnalysisCache
-        scannedDocuments:
-            ${Array.from(this.scannedDocuments.keys()).join('\n            ')}
-        analyzedDocuments:
-            ${Array.from(this.analyzedDocuments.keys()).join('\n            ')}
-      >`;
+    let result = `<AnalysisCache`;
+    if (this.scannedDocuments.size > 0) {
+      result += `
+  scannedDocuments:
+    ${Array.from(this.scannedDocuments.keys()).join('\n    ')}`;
+    }
+    if (this.analyzedDocuments.size > 0) {
+      result += `
+  analyzedDocuments:
+    ${Array.from(this.analyzedDocuments.keys()).join('\n    ')}`;
+    }
+    if (this.failedDocuments.size > 0) {
+      result += `
+  failedDocuments:
+    ${Array.from(this.analyzedDocuments.keys()).join('\n    ')}`;
```

Comment:
TFW you run with this and then you are sure you don't have a weird analysis-cache bug anymore but the files returned in toString are the wrong ones...

`this.analyzedDocuments.keys()` -> `this.failedDocuments.keys()`

Reply:
Oh man. Good catch.

That would've been a painful one to debug.